sensors/triggers/params.go

```diff
@@ -159,60 +162,71 @@ func renderEventDataAsJSON(event *v1alpha1.Event) ([]byte, error) {
-func ResolveParamValue(src *v1alpha1.TriggerParameterSource, events map[string]*v1alpha1.Event) (string, error) {
+func ResolveParamValue(src *v1alpha1.TriggerParameterSource, events map[string]*v1alpha1.Event) (*string, error) {
 	var err error
-	var value []byte
+	var eventPayload []byte
 	var key string
 	var tmplt string
-	if event, ok := events[src.DependencyName]; ok {
+
+	event, eventExists := events[src.DependencyName]
+	switch {
+	case eventExists:
 		// If context or data keys are not set, return the event payload as is
 		if src.ContextKey == "" && src.DataKey == "" && src.DataTemplate == "" && src.ContextTemplate == "" {
-			value, err = json.Marshal(&event)
+			eventPayload, err = json.Marshal(&event)
 		}
 		// Get the context bytes
 		if src.ContextKey != "" || src.ContextTemplate != "" {
 			key = src.ContextKey
 			tmplt = src.ContextTemplate
-			value, err = json.Marshal(&event.Context)
+			eventPayload, err = json.Marshal(&event.Context)
 		}
 		// Get the payload bytes
 		if src.DataKey != "" || src.DataTemplate != "" {
 			key = src.DataKey
 			tmplt = src.DataTemplate
-			value, err = renderEventDataAsJSON(event)
+			eventPayload, err = renderEventDataAsJSON(event)
 		}
-	} else if src.Value != nil {
-		return *src.Value, nil
+	case src.Value != nil:
+		// Use the default value set by the user in case the event is missing
+		return src.Value, nil
+	default:
+		// The parameter doesn't have a default value and is referencing a dependency that is
+		// missing in the received events. This is not an error and may happen with || conditions.
+		return nil, nil
 	}
+
 	if err != nil {
 		if src.Value != nil {
 			fmt.Printf("failed to parse the event data, using default value. err: %+v\n", err)
-			return *src.Value, nil
+			return src.Value, nil
 		}
-		return "", err
+		return nil, err
 	}
 	// Get the value corresponding to specified key within JSON object
-	if value != nil {
+	if eventPayload != nil {
 		if tmplt != "" {
-			out, err := getValueWithTemplate(value, tmplt)
+			out, err := getValueWithTemplate(eventPayload, tmplt)
 			if err == nil {
-				return out, nil
+				return &out, nil
 			}
 			fmt.Printf("failed to execute the src event template, falling back to key or value. err: %+v\n", err)
 		}
 		if key != "" {
-			res, err := getValueByKey(value, key)
+			res, err := getValueByKey(eventPayload, key)
 			if err == nil {
-				return res, nil
+				return &res, nil
 			}
 			fmt.Printf("Failed to get value by key: %+v\n", err)
 		}
 		if src.Value != nil {
-			return *src.Value, nil
+			return src.Value, nil
 		}
-		return string(value), nil
+
+		strPayload := string(eventPayload)
+		return &strPayload, nil
 	}
-	return "", errors.Wrapf(err, "unable to resolve '%s' parameter value", src.DependencyName)
+	return nil, errors.Wrapf(err, "unable to resolve '%s' parameter value", src.DependencyName)
```

Comment:
It seems `err` is always `nil`, shall this be changed to:

```
return nil, fmt.Errorf("unable to resolve '%s' parameter value", src.DependencyName)
```

Reply:
Yes 100%, just forgot to change this line